pkgs/development/compilers/julia/build-env.nix

```diff
@@ -0,0 +1,22 @@
+{ buildEnv
+, makeWrapper
+, extraPackages ? []
+, julia
+}:
+
+buildEnv {
+  name = "julia-env";
+  paths = extraPackages;
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  pathsToLink = [ "/share/julia" ];
+
+  postBuild = ''
+      makeWrapper ${julia}/bin/julia $out/bin/julia \
+        --set JULIA_LOAD_PATH "$JULIA_LOAD_PATH:$out/share/julia/packages" \
+        --set JULIA_DEPOT_PATH "$JULIA_DEPOT_PATH:$out/share/julia"
```

Comment:
`--suffix` doesn't work because it doesn't prepend the writable default paths.
```julia
ERROR: IOError: mkdir("/nix/store/862s5nhiqmmdyif2f5yl1xlw4arsc79b-julia-env/share/julia/logs"; mode=0o777): read-only file system (EROFS)
...
```